src/java/org/jivesoftware/openfire/archive/EmptyMessageUtils.java

```diff
@@ -0,0 +1,129 @@
+package org.jivesoftware.openfire.archive;
+
+import java.io.ByteArrayInputStream;
+
+import org.dom4j.DocumentException;
+import org.dom4j.Element;
+import org.dom4j.io.SAXReader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EmptyMessageUtils {
+
+    private static final Logger Log = LoggerFactory.getLogger(EmptyMessageUtils.class);
+
+    public enum EmptyMessageType {
+        TYPE_EVENT(-1),
+        TYPE_UNKNOWN(1),
+
+        TYPE_CHATMARKER_MARKABLE(2),  //XMLNS = urn:xmpp:chat-markers:0
+        TYPE_CHATMARKER_RECEIVED(4),
+        TYPE_CHATMARKER_DISPLAYED(8),
+        TYPE_CHATMARKER_ACKNOWLEDGED(16),  
+
+        TYPE_MESSAGE_RETRACTION(32), //XMLNS = urn:xmpp:message-retract:0
+
+        TYPE_CHATSTATE_NOTIFICATION_ACTIVE(64),  //XMLNS = http://jabber.org/protocol/chatstates'
+        TYPE_CHATSTATE_NOTIFICATION_COMPOSING(128),
+        TYPE_CHATSTATE_NOTIFICATION_PAUSED(256),
+        TYPE_CHATSTATE_NOTIFICATION_INACTIVE(512),
+        TYPE_CHATSTATE_NOTIFICATION_GONE(1024),
+
+        TYPE_MESSAGE_DELIVERY_RECEIPTS_RECEIVED(2048), //XMLNS = urn:xmpp:receipts
+        TYPE_MESSAGE_DELIVERY_RECEIPTS_REQUEST(4096);
+
+        private final long value;
+
+        EmptyMessageType(final long newValue) {
+            value = newValue;
+        }
+
+        public long getValue() { return value; }
+    }
+
+    private static Element parseWithSAX(String xmlString) throws DocumentException {
+        SAXReader xmlReader = new SAXReader();
```

Comment:
Using the default SAXReader can be dangerous, if certain properties aren't set. Using a new instance for each stanza that is processed probably requires more resources than is needed. Both issues can be prevented by using `org.jivesoftware.util.SAXReaderUtil`